templates/crane.conf.erb

```diff
@@ -5,5 +5,5 @@
 [general]
 debug: false
 #data_dir_polling_interval: 60
-data_dir: /var/www/pub/docker/app
+data_dir: /var/lib/pulp/published/docker/v2/app/
```

Comment:
Hello Katello!

So it's up to you to decide how you want to configure this, but I'll give you some more information so you can make your decision. The config as it is here will work for Docker v2 but not for Docker v1 as @ehelms and @jlsherrill pointed out. If Katello wants to continue supporting v1 content to any degree, then I would not recommend this strategy. However, if you plan to drop v1 content and go v2 only then this will work.

However, if you want to have full or partial v1 support, you will need to do something different than this. Crane's data_dir setting needs to be pointed at a directory that has app files, but it doesn't make sense to point it at /var/lib/pulp/published/docker because that folder has tons of other stuff in it that isn't for Crane. What I've been doing on my development system to support v1 and v2 is to make /var/lib/pulp/published/docker/v2/app be a symlink that points to /path/to/crane/data_dir/v2 and then I also make /var/lib/pulp/published/docker/v1/app be a similar symlink. This has the effect of pulp_docker's publish writing the app files into crane's datadir's subdirs for v1 and v2. Crane will then find the files and know what to do with them.

It isn't required that you solve the problem with symlinks like this, that is just what I have done. However you solve it, you just need to get those app files to Crane somehow. Does that help?